src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/SqsMessageReceiverTest.java

```diff
@@ -84,6 +91,20 @@ void givenCommonPlatformMessage_whenReceived_ThenProcess() throws Exception {
         verify(operation).close();
     }
 
+    @DisplayName("Given an invalid Common Platform JSON message then do not process message")
+    @Test
+    void givenInvalidCommonPlatformMessage_whenReceived_ThenDoNotProcess() throws Exception {
+        when(telemetryService.withOperation("operationId")).thenReturn(operation);
+        when(caseExtractor.extractHearing(singleCaseJson, MESSAGE_ID)).thenReturn(invalidCommonPlatformHearing);
+
+        sqsMessageReceiver.receive(singleCaseJson, MESSAGE_ID, "operationId");
+
+        verify(telemetryService).withOperation("operationId");
+        verify(telemetryService).trackHearingMessageReceivedEvent(MESSAGE_ID);
+        verifyNoInteractions(caseProcessor); //Common Platform hearing with no defendants should not be processed
+        verify(operation).close();
+    }
+
```

Comment:
It would be great to see a test like this for the isYouth filter which is added in this PR

Reply:
Added integration test